src/probnum/filtsmooth/gaussfiltsmooth/kalmanposterior.py

```diff
@@ -60,77 +68,42 @@ def __call__(self, t):
         :obj:`RandomVariable`
             Estimate of the states at time ``t``.
         """
+
+        # Recursive evaluation (t can now be any array, not just length 1)
         if not np.isscalar(t):
-            # recursive evaluation (t can now be any array, not just length 1!)
             return _RandomVariableList([self.__call__(t_pt) for t_pt in np.asarray(t)])
```

Comment:
The `np.asarray(t)` looks weird actually, we should be able to just loop over `t`.

Reply:
True. I remember that I did this for a reason (it was me who wrote this recursion, right?) but since I cannot recall why, and the tests pass just fine, I removed it.